eduNEXT/platform-plugin-turnitin
# feat: add turnitin api

### Description
This plugin moves all implementation in the Turnitin Xblock to an API in the plugin. 

### How to Test
1. Clone this repo and checkout to this branch.
2. Install the plugin in your environment.
3. Add these settings to your environment:

    ```yaml
    name: turnitin-settings
    version: 0.1.0
    patches:
      openedx-common-settings: |
        TURNITIN_TII_API_URL = "<YOUR-API-URL>"
        TURNITIN_TCA_INTEGRATION_FAMILY = "MySweetLMS"
        TURNITIN_TCA_INTEGRATION_VERSION = "3.2.4"
        TURNITIN_TCA_API_KEY = "<YOUR-API-KEY>"
    ``` 
4. Test the endpoints. You can use this Postman collection: [plugin-turnitin.postman_collection.json](https://github.com/eduNEXT/platform-plugin-turnitin/files/14100883/plugin-turnitin.postman_collection.json). The endpoint flow you should follow is:
    - Create Access Token
    - Upload File
    - Get Submission
    - Generate Similarity Report
    - Get Similarity Report
    - Get Viewer URL
